src/Client/MultiplexedConnections.cpp

```diff
@@ -126,6 +126,10 @@ void MultiplexedConnections::sendQuery(
 
     Settings modified_settings = settings;
 
+    // Kusto and PRQL queries to replicas are transformed to ClickHouse-SQL. Ensure the setting before sending.
```

Comment:
I had the change that was made in https://github.com/ClickHouse/ClickHouse/pull/59674/commits/01e203d057a37bdde4914f64bb3e86dffd022c55 in an earlier commit, and ultimately changed it to what I had because of backwards compatibility tests throwing a `Unknown setting dialect`.

I definitely agree that the generic form of blanket setting the dialect is the better way to go.